nixos/modules/services/web-apps/immich-kiosk.nix

```
@@ -0,0 +1,165 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.immich-kiosk;
+  format = pkgs.formats.yaml { };
```

... though you'll have to use
```suggestion
  format = pkgs.formats.json { };
```
for that.